base_layer/key_manager/src/mnemonic.rs

```diff
@@ -219,14 +234,19 @@ pub fn to_bytes(mnemonic_seq: &[String]) -> Result<Vec<u8>, MnemonicError> {
 /// 1) the first output 'a' is last 8 bits from input 'A', we have leftover 3 bits from 'A'
 /// 2) We add 5 bits from 'B' to generate 'b', the leftover is 6 bits from 'B'
 /// 3) We add 2 bits from 'C to generate 'c', now we have 8 bits needed to generate 'd' and we have 1 bit leftover.
-pub fn to_bytes_with_language(mnemonic_seq: &[String], language: &MnemonicLanguage) -> Result<Vec<u8>, MnemonicError> {
+pub fn to_bytes_with_language(mnemonic_seq: &SeedWords, language: &MnemonicLanguage) -> Result<Vec<u8>, MnemonicError> {
     const MASK: u64 = (1u64 << 8) - 1;
     let mut bytes = Vec::new();
     let mut rest = 0u64;
     let mut rest_bits: u8 = 0;
 
-    for curr_word in mnemonic_seq {
-        let index = find_mnemonic_index_from_word(curr_word, *language)? as u64;
+    for curr_ind in 0..mnemonic_seq.len() {
+        let index = find_mnemonic_index_from_word(
```

Comment:
Is this sensitive data?

Reply:
I wrap it with the `Hidden` type, even though these values live on the Stack. I wonder also if casting using `as` does a copy or not.

Reply:
I think `index` and the subsequent operations on it will occur in CPU registers so not concerned about that.
However, the resulting `Vec<u8>` is sensitive data

Reply:
That's cool ! Less allocations as well :) thanks @sdbondi